src/validation.cpp

```diff
@@ -3022,11 +3028,15 @@ bool ContextualCheckBlockHeader(const CBlockHeader& block, CValidationState& sta
     // Reject outdated version blocks when 95% (75% on testnet) of the network has upgraded:
     // check for version 2, 3 and 4 upgrades
     // Dogecoin: Version 2 enforcement was never used
-    if((block.GetBaseVersion() < 3 && nHeight >= consensusParams.BIP66Height) ||
-       (block.GetBaseVersion() < 4 && nHeight >= consensusParams.BIP65Height))
+    if((block.GetBaseVersion() < 3 && nHeight >= consensusParams.BIP66Height))
             return state.Invalid(false, REJECT_OBSOLETE, strprintf("bad-version(0x%08x)", block.GetBaseVersion()),
                                  strprintf("rejected nVersion=0x%08x block", block.GetBaseVersion()));
 
+    // Dogecoin: Introduce supermajority rules for v4 blocks
+    if (block.nVersion < 4 && IsSuperMajority(4, pindexPrev, consensusParams.nMajorityRejectBlockOutdated, consensusParams))
```

Comment:
Shouldn't this use `block.GetBaseVersion()`?

Reply:
Good question - master doesn't https://github.com/dogecoin/dogecoin/blob/master/src/main.cpp . Going to change that, but suspect we haven't been enforcing v3 properly from the looks of things and were saved by AuxPoW.

Reply:
Yeah, master doesn't use them in that function at all 🤔